---
title: "Poisson_fSuSiE"
author: "William R.P. Denault"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Introduction to Poisson-fSuSiE}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r , include=FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.width = 5,
  fig.height = 3,
  fig.align = "center",
  fig.cap = "&nbsp;",
  dpi = 175
  )
```


```{r setup, include=FALSE}
rm(list = ls())
library(mvPoisVA)
library(fsusieR)
library(susieR)
library(ebnm)
 load("../data/another_example.RData")
```

These data are based on sample real GTeX RNAseq counts where we model the effect of 
two SNPs on these count using Binomial thinning.

Those are simulated data of structured counts $Y$ each row being the observed "profile"
for an individual and in $X$ we store the genotype data, among those the  effect SNPs are 

```{r}
true_pos=example$true_pos
```
Here a typical count profile

```{r}
plot(example$Y[1,], main="RNAseq count for individual 1 over the region of interest")
```



The effect of the SNPs on the count are

```{r}
par(mfrow=c(1,2))
plot(example$lf[[1]], type="l", main = "effect of first causal SNPS")
plot(example$lf[[2]], type="l", main = "effect of second causal SNPS")

par(mfrow=c(1,1))

```


A simple to handle count for fine mapping is to use $log(x+1)$ transform or the Haar-Fisz transform 
when dealing with structured sequence of count as we are doing

```{r}
X=example$X
Y=example$Y
res0 = susiF(X=X, Y=log1p(Y))
res1= susiF(X=X,Y=HFT(Y))
```

Unfortunately those results are not quite correct

```{r}
res0$cs
res1$cs
true_pos
```



 

```{r}

res_poisF = Pois_fSuSiE(Y=Y,X=X ,L=3 )

```

Fine mapping the effect

```{r}
res01 <-acc_Pois_fSuSiE2 (Y=Y,X=X, L=3, post_processing = "TI"  )
res01$susiF.obj$cs
```
Estimated effect

```{r} 

plot ( res01$susiF.obj$fitted_func[[2]]  , type="l", main=paste ( "effect of SNP",true_pos[2]), col= "green4")
lines(lf[[2]])

plot ( res01$susiF.obj$fitted_func[[1]]    , type="l", main=paste ( "effect of SNP",true_pos[2]), col= "green4")
lines(lf[[1]] )
```



```{r}
res02 <-fsusieR::susiF (Y=log(Y+1),X=X, L=3, post_processing = "TI"  )
res02$susiF.obj$cs
```

```{r}

plot ( res02$ fitted_func[[2]]  , type="l", main=paste ( "effect of SNP",true_pos[2]), col= "green4")
lines(lf[[2]])

plot ( res02 $fitted_func[[1]]    , type="l", main=paste ( "effect of SNP",true_pos[2]), col= "green4")
lines(lf[[1]] )
```
